andes/core/model/model.py

```diff
@@ -546,17 +546,43 @@ def alter(self, src, idx, value):
             The device to alter
         value : float
             The desired value
+        attr : str
+            The attribute to alter, default is 'v'.
+
+        Notes
+        -----
+        New in version 1.9.3: Added the `attr` parameter and the feature to alter
+        specific attributes. This feature is useful when you need to manipulate parameter
+        values in the system base and ensure that these changes are reflected in the
+        dumped case file.
+
+        Examples
+        --------
+        >>> import andes
+        >>> ss = andes.load(andes.get_case('5bus/pjm5bus.xlsx'), setup=True)
+        >>> ss.GENCLS.alter(src='M', idx=2, value=1, attr='v')
+        >>> ss.GENCLS.get(src='M', idx=2, attr='v')
+        3.0
+        >>> ss.GENCLS.alter(src='M', idx=2, value=1, attr='vin')
+        >>> ss.GENCLS.get(src='M', idx=2, attr='v')
+        1.0
         """
 
         instance = self.__dict__[src]
 
-        if hasattr(instance, 'vin') and (instance.vin is not None):
-            self.set(src, idx, 'vin', value)
+        if not hasattr(instance, 'vin'):
+            raise AttributeError(f"{self.class_name}.{src} has no `vin` attribute.")
```

Comment:
Maybe I can make a small change there. If there is no `vin`, fall back to `v` rather than raising an error.